aten/src/ATen/native/ConstantPadNd.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ATen/ATen.h"
+
+namespace at { namespace native {
+
+Tensor constant_pad_nd(const Tensor& input, IntList pad, Scalar value) {
+    auto input_sizes = input.sizes();
+    auto l_inp = input.dim();
+
+    auto l_pad = pad.size() / 2;
+    auto l_diff = l_inp - l_pad;
+    AT_CHECK(l_inp >= l_pad, "Padding length too large");
+
+    std::vector<int64_t> new_shape;
+
+    for (int i = 0; i < l_diff; i ++) {
+        new_shape.push_back(input_sizes[i]);
+    }
+
+    for (int i = 0; i < l_pad; i++) {
+        auto pad_idx = pad.size() - ((i + 1) * 2);
+        auto new_dim = input_sizes[l_diff + i] + pad[pad_idx] + pad[pad_idx + 1];
+        AT_CHECK(new_dim > 0, "input is too small");
+        new_shape.push_back(new_dim);
+    }
+
+    auto output = at::empty(new_shape, input.options());
+    output.fill_(value);
+
+    auto c_input = input;
+    for (int i = l_diff; i < l_inp; i++) {
+        auto pad_idx = pad.size() - (i - l_diff + 1) * 2;
+        if (pad[pad_idx] < 0) {
+            c_input = c_input.narrow(i, -pad[pad_idx], c_input.sizes()[i] + pad[pad_idx]);
+        }
+        if (pad[pad_idx + 1] < 0) {
+            c_input = c_input.narrow(i, 0, c_input.sizes()[i] + pad[pad_idx + 1]);
+        }
+    }
+
+    auto c_output = output;
+    for (int i = l_diff; i < l_inp; i++) {
+        auto pad_idx = pad.size() - (i - l_diff + 1) * 2;
+        if (pad[pad_idx] > 0) {
+            c_output = c_output.narrow(i, pad[pad_idx], c_output.sizes()[i] - pad[pad_idx]);
+        }
+        if (pad[pad_idx + 1] > 0) {
+            c_output = c_output.narrow(i, 0, c_output.sizes()[i] - pad[pad_idx + 1]);
+        }
+    }
+    c_output.copy_(c_input);
```

Comment:
`copy`ing `c_input` into `c_output` will remove changes made on `c_output`. Are you sure this is what you were looking for?

Reply:
Calling `copy_` is what was done in the python version of ConstantPadNd: https://github.com/pytorch/pytorch/blob/master/torch/nn/_functions/padding.py. If you look at what happens to `c_output`, the only changes made are to its dimensions (with `narrow`). So I think it's ok that all the values are replaced.